email-smtp/src/main/java/io/commercetools/sunrise/email/smtp/SmtpAuthEmailSender.java

```diff
@@ -1,27 +1,242 @@
+     * @param port              the port number at which the SMTP server is listening for incoming connections
+     * @param transportSecurity how to secure the connection to the SMTP server
+     * @param timeoutMs         the timeout for creating, reading from and writing to SMTP connections in
+     *                          milliseconds, see {@link #SmtpAuthEmailSender(Executor, String, int, TransportSecurity, String, String, int)}
+     *                          for details
+     * @return the properties to configure the {@link Session} used by this sender
+     */
+    protected Properties createProperties(@Nonnull final String host, final int port,
+                                          @Nonnull final TransportSecurity transportSecurity,
+                                          final int timeoutMs) {
+        final Properties properties = new Properties();
+        if (System.getProperty("mail.debug") != null)
+            properties.setProperty("mail.debug", System.getProperty("mail.debug"));
+        properties.setProperty("mail.smtp.host", host);
+        properties.setProperty("mail.smtp.port", "" + port);
+        properties.setProperty("mail.smtp.auth", "" + true);
+        properties.setProperty("mail.smtp.connectiontimeout", "" + timeoutMs);
+        properties.setProperty("mail.smtp.timeout", "" + timeoutMs);
+        properties.setProperty("mail.smtp.writetimeout", "" + timeoutMs);
+        if (transportSecurity == TransportSecurity.SSL_TLS) {
+            properties.setProperty("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+            properties.setProperty("mail.smtp.socketFactory.fallback", "false");
+            properties.setProperty("mail.smtp.socketFactory.port", "" + port);
+            properties.setProperty("mail.smtp.ssl.checkserveridentity", "" + true);
+        } else if (transportSecurity == TransportSecurity.STARTTLS) {
+            properties.setProperty("mail.smtp.starttls.enable", "" + true);
+            properties.setProperty("mail.smtp.starttls.required", "" + true);
+            properties.setProperty("mail.smtp.ssl.checkserveridentity", "" + true);
+        } else if (transportSecurity == TransportSecurity.None) {
+            // No additional configuration required
+        } else {
+            throw new IllegalArgumentException("Unknown transport security: " + transportSecurity.name());
+        }
+        return properties;
     }
 
     @Override
     @Nonnull
-    public CompletionStage<String> send(@Nonnull final Consumer<MimeMessage> messageAuthor) {
-        // TODO: Optionally specify the Executor when creating the service
-        return CompletableFuture.supplyAsync(() -> "");
+    public CompletionStage<String> send(@Nonnull final MessageEditor messageEditor) {
+        final MimeMessage message = createAndFillMessage(messageEditor);
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
Yes.

The Java 8 async stuff isn't exactly well designed and documented.
I think it is way to go since it is not so restrictive concerning the error handling and transportation.